packages/core/src/browser/dialogs.ts

```diff
@@ -277,7 +292,7 @@ export class ConfirmDialog extends AbstractDialog<boolean> {
         @inject(ConfirmDialogProps) protected readonly props: ConfirmDialogProps
     ) {
         super(props);
-
+        this.id = 'theia-confirm-dialog';
```

Comment:
Can you explain why providing an `id` is necessary in the scope of the PR?
It'd probably be better to add a new `prop` where extenders can provide custom `ids`.

https://github.com/theia-ide/theia/blob/cc51575a2c8660f13794ed9099ee14daaca982b9/packages/core/src/browser/dialogs.ts#L23-L25

Reply:
Because I'm going to use this ID to determine if it's the current component.